File: go-gateway/internal/handler/health.go
package handler

import (
	"net/http"
	"time"

	"credit-card-service/go-gateway/internal/response"
)

type HealthHandler struct{}

func NewHealthHandler() *HealthHandler {
	return &HealthHandler{}
}

func (h *HealthHandler) HealthCheck(w http.ResponseWriter, r *http.Request) {
	response.JSON(w, http.StatusOK, map[string]interface{}{
		"status":    "ok",
		"service":   "go-gateway",
		"router":    "chi/v5",
		"timestamp": time.Now().UTC().Format(time.RFC3339),
	})
}

File: fastapi-agent/app/api/health.py
from datetime import datetime, timezone
from fastapi import APIRouter

router = APIRouter()


@router.get("/health", tags=["Health"])
def health_check():
    return {
        "status": "ok",
        "service": "fastapi-agent",
        "architecture": "modular",
        "timestamp": datetime.now(timezone.utc).isoformat(),
    }
